go-controller/pkg/util/gateway-init.go

```diff
@@ -395,3 +400,181 @@ func GatewayInit(clusterIPSubnet []string, nodeName, ifaceID, nicIP, nicMacAddre
 
 	return nil
 }
+
+// LocalGatewayInit creates a gateway router to access local service.
+func LocalGatewayInit(clusterIPSubnet []string, nodeName, ifaceID, nicIP, nicMacAddress,
+	defaultGW string, physnetName string) error {
+
+	ip, physicalIPNet, err := net.ParseCIDR(nicIP)
+	if err != nil {
+		return fmt.Errorf("error parsing %s (%v)", nicIP, err)
+	}
+	n, _ := physicalIPNet.Mask.Size()
+	physicalIPMask := fmt.Sprintf("%s/%d", ip.String(), n)
+	physicalIP := ip.String()
+
+	if defaultGW != "" {
+		defaultgwByte := net.ParseIP(defaultGW)
+		defaultGW = defaultgwByte.String()
+	}
+
+	k8sClusterRouter, err := GetK8sClusterRouter()
+	if err != nil {
+		return err
+	}
+
+	systemID, err := GetNodeChassisID()
+	if err != nil {
+		return err
+	}
+
+	// Create a gateway router.
+	gatewayRouter := "GR_local_" + nodeName
+	stdout, stderr, err := RunOVNNbctl("--", "--may-exist", "lr-add",
+		gatewayRouter, "--", "set", "logical_router", gatewayRouter,
+		"options:chassis="+systemID, "external_ids:physical_ip="+physicalIP)
+	if err != nil {
+		return fmt.Errorf("Failed to create logical router %v, stdout: %q, "+
+			"stderr: %q, error: %v", gatewayRouter, stdout, stderr, err)
+	}
+
+	gwSwitchPort := "jtor-" + gatewayRouter
+	gwRouterPort := "rtoj-" + gatewayRouter
+	routerMac, routerCIDR, err := ensureGatewayPortAddress(gwSwitchPort)
+	if err != nil {
+		return err
+	}
+
+	// Must move the IP from the LSP to the LRP and set the LSP addresses
+	// to 'router' in one transaction, because IPAM doesn't consider LSPs that
+	// are attached to routers when checking reserved addresses.
+	stdout, stderr, err = RunOVNNbctl(
+		"--", "--may-exist", "lrp-add", gatewayRouter, gwRouterPort, routerMac.String(), routerCIDR.String(),
+		"--", "set", "logical_switch_port", gwSwitchPort, "type=router",
+		"options:router-port="+gwRouterPort, "addresses=router")
+	if err != nil {
+		return fmt.Errorf("failed to add logical port to router, stdout: %q, "+
+			"stderr: %q, error: %v", stdout, stderr, err)
+	}
+
+	// When there are multiple gateway routers (which would be the likely
+	// default for any sane deployment), we need to SNAT traffic
+	// heading to the logical space with the Gateway router's IP so that
+	// return traffic comes back to the same gateway router.
+	stdout, stderr, err = RunOVNNbctl("set", "logical_router",
+		gatewayRouter, "options:lb_force_snat_ip="+routerCIDR.IP.String())
+	if err != nil {
+		return fmt.Errorf("Failed to set logical router, stdout: %q, "+
+			"stderr: %q, error: %v", stdout, stderr, err)
+	}
+
```

Comment:
L468 to 470 is not required for use case.. since it is for ExternalIP usecase.... so you can delete